Source/carthage/Extensions.swift

```diff
@@ -47,6 +47,28 @@ internal func print<T>(object: T) {
 	}
 }
 
+internal class FileHandle {
+	let fileDescriptor: Int32
+	let pointer: UnsafeMutablePointer<FILE>
+
+	init(fileDescriptor fd: Int32) {
+		fileDescriptor = fd
+		pointer = fdopen(fd, "r+")
```

Comment:
> Should this be a failable initializer then?

I think we should do an explicit `open` API. I’d like to hear @jspahrsummers’ thoughts on this, tho; this may be considered sufficient for Carthage’s needs, and he’s in a better place to estimate that than I am.

> What should the client do if a FileHandle is repeatedly unable to be initialized?

That’s pretty much up to the client.